common/events/platform/boost/events.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <common/events.hpp>
+
+#include <boost/asio.hpp>
+
+namespace asio = boost::asio;
+
+namespace mender {
+namespace events {
+
+void EventLoop::Run() {
+	ctx_.run();
+}
+
+void EventLoop::Stop() {
+	ctx_.stop();
+}
+
+Timer::Timer(EventLoop &loop) :
+	timer_(loop.ctx_) {
+}
+
+template <typename Duration>
+void Timer::Wait(Duration duration) {
+	timer_.expires_after(duration);
+	timer_.wait();
+}
+template void Timer::Wait<std::chrono::milliseconds>(std::chrono::milliseconds duration);
```

Comment:
Q:
Why template the function definition and explicitly initiate it here, as opposed to having two overloaded functions?

Reply:
As mentioned f2f, I was conflicted about this decision. My main motivation was to try to keep it out of the header file, and only in the source file. Which is tidy, but comes with the rather significant downside that you can only use the exact types you defined. I thought maybe this was ok for seconds and milliseconds.

But now that you have to move some of your definitions into header files as well, I will just do the same, there is no reason to keep it like this.